pallets/msa/src/lib.rs

```diff
@@ -159,6 +185,33 @@ impl<T: Config> Pallet<T> {
 
 		Ok(())
 	}
+
+	pub fn revoke_key(key: &T::AccountId) -> DispatchResult {
+		KeyInfoOf::<T>::try_mutate(key, |maybe_info| -> DispatchResult {
+			let mut info = maybe_info.take().ok_or(Error::<T>::NoKeyExist)?;
+
+			ensure!(info.expired == T::BlockNumber::zero(), Error::<T>::KeyRevoked);
+
+			let current_block = frame_system::Pallet::<T>::block_number();
```

Comment:
We don't want to allow a block number to be specified for the revocation?

Reply:
I think so.